File: dualpal.py
"""
ID: totheso1
LANG: PYTHON3
TASK: dualpal
NOTE: Actually, there exists a faster algorithm, which construct every palindrome for each base for small ones to larger ones, and use heap to track the least two of them to see if they are equal. But it might not worth to be implemented.
"""
from typing import List
from itertools import islice

fin = open("dualpal.in", "r")
fout = open("dualpal.out", "w")


def fprint(*args, **kwargs) -> None:
    print(*args, file=fout, **kwargs)


# Code start
def base10_to_baseN(number: int, base: int) -> List[int]:
    # if not 2 <= base <= 10:
    #     raise ValueError("Convertion base should settle between 2 to 10 inclusively.")
    res = []
    while number:
        res.append(number % base)
        number //= base
    return res


def is_pal(digits: List[int]) -> bool:
    return all(
        left == right
        for left, right in zip(
            islice(digits, len(digits) // 2), islice(reversed(digits), len(digits) // 2)
        )
    )


def check_dual_pal(number: int) -> bool:
    return sum(is_pal(base10_to_baseN(number, base)) for base in range(2, 11)) >= 2


N, S = map(int, fin.read().split())

S += 1
while N != 0:
    if check_dual_pal(S):
        fprint(S)
        N -= 1
    S += 1
# Code end

File: dualpal.cpp
/*
ID: totheso1
LANG: C++14
TASK: dualpal
*/

#include <fstream>
#include <iterator>
#include <vector>

using namespace std;

static ifstream fin("dualpal.in");
static ofstream fout("dualpal.out");

#define endl '\n'

template <typename T>
T fin_get() {
  return *istream_iterator<T>(fin);
}

template <typename C>
C fin_get_collection(int size) {
  C ret;
  copy_n(istream_iterator<typename C::value_type>(fin), size,
         back_inserter(ret));
  return ret;
}

static vector<int> base10_to_baseN(int num, const int base) {
  vector<int> ret;
  while (num) {
    ret.push_back(num % base);
    num /= base;
  }
  return ret;
}

static bool is_pal(const vector<int> &digits) {
  return equal(cbegin(digits), cbegin(digits) + digits.size() / 2, crbegin(digits));
}

static bool check_dual_pal(const int num) {
  int sum = 0;
  for (int base = 2; base <= 10; ++base) {
    if (is_pal(base10_to_baseN(num, base))) {
      ++sum;
      if (sum == 2) {
        return true;
      }
    }
  }
  return false;
}

int main() {
  const int N = fin_get<int>();
  const int S = fin_get<int>();

  for (int num = S + 1, remain = N; remain > 0; ++num) {
    if (check_dual_pal(num)) {
      fout << num << endl;
      --remain;
    }
  }
}
